### PR TITLE
Deployment on-boarding page

### PR DESCRIPTION
**Pull request by Krishanx92**
Screenshot from 2021-04-07 17-38-22
Screenshot from 2021-04-07 17-38-48

Fix https://github.com/wso2/product-apim/issues/10705
Fix https://github.com/wso2/product-apim/issues/10814

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx [hunk 1]
```diff
@@ -0,0 +1,312 @@
+import React, { useState } from 'react';
+import 'react-tagsinput/react-tagsinput.css';
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx:2]: Why do we need tag input here?

**Review by tmkasun — CHANGES_REQUESTED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    const classes1 = useStyles();
+    const theme = useTheme();
+    const { maxCommentLength } = theme.custom;
+    const { settings } = useAppContext();
+    const defaultVhosts = settings.environment.map(
+        (e) => (e.vhosts && e.vhosts.length > 0 ? { env: e.name, vhost: e.vhosts[0].host } : undefined),
+    );
+    const [DescriptionOpen, setDescriptionOpen] = useState(false);
+    const [SelectedEnvironment, setSelectedEnvironment] = useState([]);
+    // const [currentLength, setCurrentLength] = useState(0);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx:82]: Shall we remove this comment ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    const [selectedVhostDeploy, setVhostsDeploy] = useState(defaultVhosts);
+
+    const handleDescriptionOpen = () => {
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx:85]: This has only 1 usage, Shall we remove and use lambda function directly in event handler

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        if (event.target.checked) {
+            setSelectedEnvironment([...SelectedEnvironment, event.target.value]);
+        } else {
+            setSelectedEnvironment(
+                SelectedEnvironment.filter((env) => env !== event.target.value),
+            );
+        }
+        if (event.target.name === 'description') {
+            setDescription(event.target.value);
+            // setCurrentLength(event.target.value.length);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx:105]: Shall we remove this commented line

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+                                                    .description.deploy'
+                                                    defaultMessage='Brief description of the new revision'
+                                                />
+                                            )}
+                                            multiline
+                                            rows={3}
+                                            defaultValue={description === true ? '' : description}
+                                            onBlur={handleChange}
+                                        />
+                                        {/* <Typography className={classes.textCount} align='right'>
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx:277]: Shall we remove this commented line ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+    } = props;
+    const classes1 = useStyles();
+    const theme = useTheme();
+    const { maxCommentLength } = theme.custom;
+    const { settings } = useAppContext();
+    const defaultVhosts = settings.environment.map(
+        (e) => (e.vhosts && e.vhosts.length > 0 ? { env: e.name, vhost: e.vhosts[0].host } : undefined),
+    );
+    const [DescriptionOpen, setDescriptionOpen] = useState(false);
+    const [SelectedEnvironment, setSelectedEnvironment] = useState([]);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx:81]: Normally, only React component names start with capital letters, so shall we make this `S` lower case?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx (hunk 1 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx]: Fixed https://github.com/wso2/carbon-apimgt/pull/10270/commits/ee20a3b9b54f0df2bafd12c6a3e622938b22935e

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx (hunk 2 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx]: Fixed https://github.com/wso2/carbon-apimgt/pull/10270/commits/ee20a3b9b54f0df2bafd12c6a3e622938b22935e

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx (hunk 3 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx]: Fixed https://github.com/wso2/carbon-apimgt/pull/10270/commits/ee20a3b9b54f0df2bafd12c6a3e622938b22935e

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx (hunk 4 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx]: Fixed https://github.com/wso2/carbon-apimgt/pull/10270/commits/ee20a3b9b54f0df2bafd12c6a3e622938b22935e

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx (hunk 5 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx]: Fixed https://github.com/wso2/carbon-apimgt/pull/10270/commits/ee20a3b9b54f0df2bafd12c6a3e622938b22935e

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx (hunk 6 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/DeploymentOnbording.jsx]: Fixed https://github.com/wso2/carbon-apimgt/pull/10270/commits/ee20a3b9b54f0df2bafd12c6a3e622938b22935e

**Review by tmkasun — APPROVED** [review]

**Pull request merged**